server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -2220,6 +2220,7 @@ public void run() {
       synchronized (openingTablets) {
         while (openingTablets.contains(extent)) {
           try {
+            log.info("Waiting for tablet {} to finish opening before unloading.", extent);
```

Comment:
Should this be a debug message instead of info?

Reply:
I was actually leaning towards a warning since the thread could sit here for a long time after only printing this message once.  But this could also be normal behavior in some situations so I settled on info. I don't think it should get printed a lot, unless something is wrong, hence why I was thinking of warning. This will only get printed if a tablet is being unloaded while also trying to be opened.

Reply:
If it can happen as a result of a user action that they have control over, then warning might make more sense. If it's something that happens, and is informative to all users, then info might make sense. If it's informative only for troubleshooting, then debug might make sense. I'm not sure which this falls in. I assumed the "only for troubleshooting" category, but I haven't dived in to form a more informed opinion.

Reply:
I am working on a change in this code for 2.1 and it includes a lot more logging, like this one and they all seem to be more appropriate at debug.  I can make it debug since its really out of user control.

I have noticed something strange with this class logging in 2.1 that I can't seem to figure out.  I don't see any of the logging for the UnloadTabletHandler in 2.1 (using Uno) not even any of the INFO logs.  I wonder if its because the UnloadTabletHandler got moved out of TabletServer class when the ThriftClientHandler was moved out in #1584.  I also wonder if any of the other classes in that change were affected.

Reply:
Now that I think about it, I think I ran into the same thing regarding logging in ThriftClientHandler. I'll have to double-check tomorrow.

Reply:
@milleruntime I looked at all the current logging in UnloadTabletHandler, and they all look like unlikely edge cases. I wouldn't be surprised if we rarely saw logs from it. If you add new logging to it that should show up, and it still doesn't, then it might be a problem... or if you know for sure one of these edge cases happened but it didn't get logged.

Reply:
@ctubbsii I agree, with the exception of what gets logged using the new `TabletLogger` Keith added in #1316 .  But I am not seeing anything from the `TabletLogger` and I am not sure why.  Maybe there is something we have to do with Log4j2 to include those logs?  Here is the class: https://github.com/apache/accumulo/blob/cccac86255cd2d0ecd49e791162ca5cdf07512b6/core/src/main/java/org/apache/accumulo/core/logging/TabletLogger.java

Reply:
@milleruntime Are you sure you're checking for the correct logger names? The logger names should look like `tablet.X`.

I just quickly ran Uno and saw them with `grep '\[tablet[.][a-z]' *`
```
manager_localhost.log:2021-01-26T16:33:51,850 [tablet.location] DEBUG: Assigned +r<< to localhost:9997[100032100b90004]
manager_localhost.log:2021-01-26T16:33:54,498 [tablet.location] DEBUG: Assigned !0;~< to localhost:9997[100032100b90004]
...
tserver_localhost.log:2021-01-26T16:34:21,075 [tablet.files] DEBUG: Flushed !0;~< created hdfs://localhost:8020/accumulo/tables/!0/table_info/F0000000.rf from [memory]
tserver_localhost.log:2021-01-26T16:34:21,271 [tablet.files] DEBUG: Flushed +r<< created hdfs://localhost:8020/accumulo/tables/+r/root_tablet/F0000001.rf from [memory]
```

Remember, the `TabletLogger` loggers don't use the class name for the logger, they have custom logger names.

Reply:
If I run `grep '\[tablet[.][a-z]' ` I do see some...
<pre>
tserver1_ip-10-113-12-25.log:2021-01-26T16:03:45,008 [tablet.files] DEBUG: Flushed +r<< created hdfs://localhost:8020/accumulo/tables/+r/root_tablet/F00000ay.rf from [memory]
tserver1_ip-10-113-12-25.log:2021-01-26T16:03:45,009 [tablet.files] DEBUG: Compacting +r<< on root.small for SYSTEM from [F00000ay.rf, A00000ax.rf] size 1 KB
</pre>

But i am not seeing any for user tablets.  I have been running RW MultiTable so I have plenty of activity. Also, if I run `grep "Assigned" *` I don't see any results.

Reply:
Nevermind, I was running a build from an experimental branch I have and I guess it's not behaving correctly.  I built again from main branch and am now seeing all the logging. Sorry!